### PR TITLE
feat: added possibility to order requests alphabetically (#2261)

### LINKED ISSUE #2261: [feature]: Can I have my collection sorted alphabetically?
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

image

### Why should this be worked on?

It is hard for me to find particular entry that I need, I wish they could be sorted alphabetically.

### PR DESCRIPTION
**Pull request by AbderrahmaneGuermat**
Closes #2261 

### Description

I added a button which allows users to sort the requests alphabetically. To implement this feature I added a flag which controls the sorting process of the requests

### Checks

- [X] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [X] All the tests have passed

**Current preview of the feature**

https://user-images.githubusercontent.com/61157096/194758255-03126f84-fffc-422b-8bd4-90664a8008da.mp4

### CODE REVIEW

File: packages/hoppscotch-app/src/components/collections/index.vue [hunk 1]
```diff
@@ -955,6 +971,9 @@ export default defineComponent({
         this.displayConfirmModal(false)
       }
     },
+    ChangeFlagOrder() {
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/src/components/collections/index.vue:974]: Function names should be `camelCase`

File: packages/hoppscotch-app/src/components/collections/index.vue [hunk 2]
```diff
@@ -44,6 +44,13 @@
           @click="displayModalAdd(true)"
         />
         <span class="flex">
+          <ButtonSecondary
+            v-tippy="{ theme: 'tooltip' }"
+            blank
+            :title="Order"
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/src/components/collections/index.vue:50]: There should be an `i18n` tag here.
Also this bound variable doesn't exist.
`:prop`'s assigned value is not considered as a string, but as a JS expression.

React equivalent
```
<ButtonSecondary title={{ Order }} />
```

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
The feature is not working.

Clicking on the sort button doesn't do anything in My Collections section from my usage.

**Comment by liyasthomas** [comment]
Save context doesn't respect sorted order.

Steps to reproduce:
1. Select a request from any Collection
2. Apply sort on the collections
3. See the select/save context doesn't respect sorted order instead it sticks to the previous order which breaks the feature.

cc: @AndrewBastin

**Comment by AndrewBastin** [comment]
As there is no response from the OP in the past 10 days, this PR will now be closed because of inactivity.

We can reopen if you are interested in pursuing the PR.

**Pull request closed without merging**